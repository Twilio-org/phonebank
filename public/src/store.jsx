import { applyMiddleware, createStore, combineReducers, compose } from 'redux';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer as formReducer } from 'redux-form';

import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import immutable from 'redux-immutable-state-invariant';
import { createLogger } from 'redux-logger';

import { authStatusReducer, LOGOUT_USER } from './reducers/login';
import { accountInfoReducer } from './reducers/account_info';
import { campaignListReducer } from './reducers/campaign';
import { adminQuestionsReducer } from './reducers/admin_questions';
import { adminScriptsReducer } from './reducers/admin_scripts';
import { adminContactListsReducer } from './reducers/admin_contact_lists';

const appReducer = combineReducers({
  form: formReducer,
  auth: authStatusReducer,
  account_info: accountInfoReducer,
  admin_campaigns: campaignListReducer,
  admin_questions: adminQuestionsReducer,
  admin_scripts: adminScriptsReducer,
  admin_contact_lists: adminContactListsReducer
});

const rootReducer = (state, action) => {
  const { type } = action;
  const newState = (type === LOGOUT_USER) ? undefined : state;
  return appReducer(newState, action);
};

const middleware = [immutable(), createLogger(), promise(), thunk];

const store = createStore(rootReducer, compose(applyMiddleware(...middleware), autoRehydrate()));

persistStore(store);

// to reset the store:
// persistStore(store).purge();

export default store;
