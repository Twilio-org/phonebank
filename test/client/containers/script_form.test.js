import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ScriptNewFormContainer from '../../../public/src/containers/create_script';
import store from '../../../public/src/store';
import { exposeLocalStorageMock } from '../client_test_helpers';

exposeLocalStorageMock();

describe('<ScriptNewFormContainer />', () => {
  describe('rendering', () => {
    const routerWrappedForm = (
      <Provider store={store}>
        <MemoryRouter>
          <ScriptNewFormContainer />
        </MemoryRouter>
      </Provider>
    );

    const wrapper = mount(
      routerWrappedForm
    );
    it('renders correctly', () => {
      const tree = renderer.create(
        routerWrappedForm
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
