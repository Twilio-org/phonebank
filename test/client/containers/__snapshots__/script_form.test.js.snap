// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScriptNewFormContainer /> rendering renders correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <h1>
      Create Script
    </h1>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-xs-6"
      >
        <div
          className="form-group "
        >
          <label
            htmlFor="Name (internal)"
          >
            Name (internal)
            <input
              className="form-control"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Name (internal)"
              size="51"
              type="Name (internal)"
              value=""
            />
          </label>
          <div
            className="text-danger"
          >
            
          </div>
        </div>
      </div>
      <div
        className="col-xs-6"
      >
        <br />
        <p>
          This is the name that admin will see when selecting a script
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-6"
      >
        <div
          className="form-group "
        >
          <label
            htmlFor="Description (public)"
          >
            Description (public)
          </label>
          <textarea
            className="form-control"
            cols="50"
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Description (public)"
            rows="6"
            value=""
          />
          <div
            className="text-danger"
          >
            
          </div>
        </div>
      </div>
      <div
        className="col-xs-6"
      >
        <br />
        <p>
          Provides details on how to use the script.
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-6"
      >
        <div
          className="form-group "
        >
          <label
            htmlFor="Body"
          >
            Body
          </label>
          <textarea
            className="form-control"
            cols="50"
            name="body"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Body"
            rows="6"
            value=""
          />
          <div
            className="text-danger"
          >
            
          </div>
        </div>
      </div>
      <div
        className="col-xs-6"
      >
        <br />
        <p>
          Script text that volunteer will read before asking questions.
        </p>
      </div>
    </div>
    <div
      className="page-header"
    >
      <h1>
        Assign Script Questions
      </h1>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-6"
      >
        <div
          className="row form-group "
        >
          <label
            htmlFor="dropdown-basic-1"
          >
            Question 1:
          </label>
          <select
            className="form-control"
            id="dropdown-basic-1"
            name="question1"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option>
              Select 
              Question 1:
            </option>
          </select>
        </div>
        <div
          className="row form-group "
        >
          <label
            htmlFor="dropdown-basic-2"
          >
            Question 2:
          </label>
          <select
            className="form-control"
            id="dropdown-basic-2"
            name="question2"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option>
              Select 
              Question 2:
            </option>
          </select>
        </div>
        <div
          className="row form-group "
        >
          <label
            htmlFor="dropdown-basic-3"
          >
            Question 3:
          </label>
          <select
            className="form-control"
            id="dropdown-basic-3"
            name="question3"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option>
              Select 
              Question 3:
            </option>
          </select>
        </div>
        <div
          className="row form-group "
        >
          <label
            htmlFor="dropdown-basic-4"
          >
            Question 4:
          </label>
          <select
            className="form-control"
            id="dropdown-basic-4"
            name="question4"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option>
              Select 
              Question 4:
            </option>
          </select>
        </div>
        <div
          className="row form-group "
        >
          <label
            htmlFor="dropdown-basic-5"
          >
            Question 5:
          </label>
          <select
            className="form-control"
            id="dropdown-basic-5"
            name="question5"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option>
              Select 
              Question 5:
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-4"
      >
        <button
          className="btn btn-success"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Create New Question
        </button>
      </div>
      <div
        className="col-xs-6"
      >
        <div
          className="btn-toolbar"
          role="toolbar"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            Save Script
          </button>
          <button
            className="btn btn-default"
            disabled={false}
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
